action.yml

```
@@ -168,21 +163,141 @@ runs:
+                exit 1
+              fi
+              . __nvm_install.sh
+              rm __nvm_install.sh
+
+              # Load nvm without needing terminal restart
+              export NVM_DIR="$HOME/.nvm"
+              [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+              [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
+            fi
+
+            # Install correct version of node and yarn
+            nvm install $NVM_VERSION
+            nvm use $NVM_VERSION
+            if [[ $(which yarn) == "" ]]; then
+              npm install -g yarn
+            fi
+
+            # Get composer name of the module
+            COMPOSER_NAME=$(jq -r ".name" composer.json)
+
+            # Install silverstripe/admin in sibling directory if needed so that shared components are available
+            if [[ $COMPOSER_NAME != silverstripe/admin ]]; then
+              DIR=$(pwd)
+
+              # Work out version of silverstripe/admin to checkout
+              cd ..
+              mkdir __tmp
+              cd __tmp
+              ADMIN_VERSION=$(COMPOSER_NAME=$COMPOSER_NAME INTO_BRANCH=$INTO_BRANCH php -r '
+                $COMPOSER_NAME = getenv("COMPOSER_NAME");
+                $INTO_BRANCH = getenv("INTO_BRANCH");
+                $json = [
+                  "require" => [
+                    "$COMPOSER_NAME" => "$INTO_BRANCH.x-dev",
+                    "silverstripe/recipe-kitchen-sink" => "*"
+                  ],
+                  "prefer-stable" => false,
+                  "minimum-stability" => "dev"
+                ];
+                file_put_contents("composer.json", json_encode($json, JSON_UNESCAPED_SLASHES));
+                shell_exec("composer update --no-install");
+                $lock = json_decode(file_get_contents("composer.lock"), true);
+                $version = array_values(array_filter(
+                  $lock["packages"],
+                  fn($p) => $p["name"] === "silverstripe/admin"
+                ))[0]["version"];
+                echo str_replace(".x-dev", "", $version);
+              ')
+              cd ..
+              rm -rf __tmp
+              if [[ $ADMIN_VERSION == "" ]]; then
+                echo "Unable to determine version of silverstripe/admin version to checkout"
+                exit 1
+              fi
+              echo "ADMIN_VERSION is $ADMIN_VERSION"
+
+              # Install admin if required
+              if [[ ! -d admin ]]; then
+                git clone https://github.com/silverstripe/silverstripe-admin.git admin
+              fi
+
+              # Checkout admin version
+              cd admin
+              git checkout $ADMIN_VERSION
+              yarn install
+              cd $DIR
+            fi
+
+            # Rebuild dist files
+            yarn build
+
+            # Only add client files
+            for FILE in $CLIENT_DIFF_FILES; do
+              echo "Adding $FILE to commit"
+              git add $FILE
+            done
```

```suggestion
            # Only add client dist files
            git add client/dist
```
We don't want to add non-dist files - and there may be different dist files affected across major versions, so we shouldn't rely on the diff we made before running `yarn build`.
We should instead trust that if `yarn build` is successful, then it is correct - and therefore everything in the client/dist folder is correct.

---

Updated